src/rosdiscover/observer/ros1.py

```diff
@@ -0,0 +1,135 @@
+# -*- coding: utf-8 -*-
+from typing import Callable, Collection, Dict, Mapping
+
+from loguru import logger
+from roswire import AppInstance
+
+from .observer import Observer
+from .ros1_actions import ActionCandidate, separate_topics_from_action
+from .. import Config
+from ..interpreter import NodeContext, ParameterServer, SystemSummary
+
+_NODES_TO_FILTER_OUT = {'/rosout'}
+_TOPICS_TO_FILTER_OUT = {'/rosout', '/rosout_agg'}
+_SERVICES_TO_FILTER_OUT = {'set_logger_level', 'get_loggers'}
+
+
+class ROS1Observer(Observer):
+
+    def __init__(self, app: AppInstance, config: Config):
+        super().__init__(app, config)
+
+    def observe_and_summarise(self) -> SystemSummary:
+        """
+        Takes a snapshot of the ROS system running in the container referred to by
```

Comment:
Since `self.app_instance` is an implementation detail, we don't really need to expose it here. Instead, we can just say that this method observes the instantaneous state of the associated ROS system.